os/drivers/video/video_null_driver.c

```diff
@@ -0,0 +1,1554 @@
+	{V4L2_CTRL_CLASS_USER, V4L2_CID_BLUE_BALANCE, V4L2_CTRL_TYPE_INTEGER, "Blue Balance", -50, 50, 2, 0, 0},
+	{V4L2_CTRL_CLASS_USER, V4L2_CID_GAMMA, V4L2_CTRL_TYPE_INTEGER, "Gama", 8, 30, 2, 16, 16},
+	{V4L2_CTRL_CLASS_USER, V4L2_CID_GAMMA_CURVE, V4L2_CTRL_TYPE_INTEGER, "Gama Curve", 1, 30, 1, 17, 17},
+	{V4L2_CTRL_CLASS_USER, V4L2_CID_EXPOSURE, V4L2_CTRL_TYPE_INTEGER, "Exposure", -64, 64, 1, 22, 22},
+	{V4L2_CTRL_CLASS_USER, V4L2_CID_HFLIP, V4L2_CTRL_TYPE_INTEGER, "Hflip", 0, 1000, 1, 850, 850},
+	{V4L2_CTRL_CLASS_USER, V4L2_CID_VFLIP, V4L2_CTRL_TYPE_INTEGER, "Vflip", 0, 1000, 1, 350, 350},
+	{V4L2_CTRL_CLASS_USER, V4L2_CID_HFLIP_STILL, V4L2_CTRL_TYPE_INTEGER, "Hflip Still", 9, 35, 1, 17, 17},
+	{V4L2_CTRL_CLASS_USER, V4L2_CID_VFLIP_STILL, V4L2_CTRL_TYPE_INTEGER, "Vflip Still", 1, 80, 1, 54, 54},
+	{V4L2_CTRL_CLASS_USER, V4L2_CID_SHARPNESS, V4L2_CTRL_TYPE_INTEGER, "Sharpness", -255, 255, 1, 112, 112},
+	{V4L2_CTRL_CLASS_USER, V4L2_CID_COLOR_KILLER, V4L2_CTRL_TYPE_INTEGER, "Color Killer", 0, 8, 2, 6, 6},
+	{V4L2_CTRL_CLASS_USER, V4L2_CID_COLORFX, V4L2_CTRL_TYPE_INTEGER, "ColorFX", 0, 64, 1, 22, 22},
+	{V4L2_CTRL_CLASS_USER, V4L2_CID_AUTOBRIGHTNESS, V4L2_CTRL_TYPE_INTEGER, "Auto Brightness", 0, 128, 1, 16, 16},
+	{V4L2_CTRL_CLASS_USER, V4L2_CID_ROTATE, V4L2_CTRL_TYPE_INTEGER, "Rotate", 0, 4, 1, 3, 3},
+	{V4L2_CTRL_CLASS_CAMERA, V4L2_CID_EXPOSURE_AUTO, V4L2_CTRL_TYPE_MENU, "Auto Exposure", 0, 10000, 1, 196, 196},
+	{V4L2_CTRL_CLASS_CAMERA, V4L2_CID_EXPOSURE_ABSOLUTE, V4L2_CTRL_TYPE_INTEGER, "Exposure Absolute", 0, 30, 1, 0, 0},
+	{V4L2_CTRL_CLASS_CAMERA, V4L2_CID_FOCUS_ABSOLUTE, V4L2_CTRL_TYPE_INTEGER, "Focus Absolute", -128, 128, 1, -115, -115},
+	{V4L2_CTRL_CLASS_CAMERA, V4L2_CID_FOCUS_RELATIVE, V4L2_CTRL_TYPE_INTEGER, "Focus Relative", -128, 128, 1, 96, 96},
+	{V4L2_CTRL_CLASS_CAMERA, V4L2_CID_FOCUS_AUTO, V4L2_CTRL_TYPE_BOOLEAN, "Focus Auto", -128, 128, 1, 16, 16},
+	{V4L2_CTRL_CLASS_CAMERA, V4L2_CID_ZOOM_ABSOLUTE, V4L2_CTRL_TYPE_INTEGER, "Zoom Absolute", -256, 256, 1, -136, -136},
+	{V4L2_CTRL_CLASS_CAMERA, V4L2_CID_ZOOM_RELATIVE, V4L2_CTRL_TYPE_INTEGER, "Zoom Relative", -256, 256, 1, -185, -185},
+	{V4L2_CTRL_CLASS_CAMERA, V4L2_CID_ZOOM_CONTINUOUS, V4L2_CTRL_TYPE_INTEGER, "Zoom Continuous", -256, 256, 1, 136, 136},
+	{V4L2_CTRL_CLASS_CAMERA, V4L2_CID_IRIS_ABSOLUTE, V4L2_CTRL_TYPE_INTEGER, "Iris Absolute", 0, 30, 2, 28, 28},
+	{V4L2_CTRL_CLASS_CAMERA, V4L2_CID_IRIS_RELATIVE, V4L2_CTRL_TYPE_INTEGER, "Iris Relative", 0, 30, 2, 26, 26},
+	{V4L2_CTRL_CLASS_CAMERA, V4L2_CID_AUTO_N_PRESET_WHITE_BALANCE, V4L2_CTRL_TYPE_INTEGER, "White Balance(Preset)", -1, 1, 1, 1, 1},
+	{V4L2_CTRL_CLASS_CAMERA, V4L2_CID_WIDE_DYNAMIC_RANGE, V4L2_CTRL_TYPE_INTEGER, "Wide Dynamic Range", 0, 40, 1, 19, 19},
+	{V4L2_CTRL_CLASS_CAMERA, V4L2_CID_IMAGE_STABILIZATION, V4L2_CTRL_TYPE_INTEGER, "Image Stabilization", 0, 64, 1, 45, 45},
+	{V4L2_CTRL_CLASS_CAMERA, V4L2_CID_ISO_SENSITIVITY, V4L2_CTRL_TYPE_INTEGER, "Iso Sensitivity", 0, 128, 1, 101, 101},
+#if 0
```

Comment:
Could you let me know why you keep #if 0?

Reply:
Done